core/services/ocr2/plugins/threshold/decryption_queue_test.go

```diff
@@ -0,0 +1,357 @@
+
+	ctx1, cancel1 := context.WithTimeout(context.Background(), time.Duration(1000)*time.Millisecond)
+	defer cancel1()
+
+	//nolint:errcheck
+	go dq.Decrypt(ctx1, []byte("11"), []byte("encrypted"))
+
+	waitForPendingRequestToBeAdded(t, dq, []byte("11"))
+
+	ctx2, cancel2 := context.WithTimeout(context.Background(), time.Duration(1000)*time.Millisecond)
+	defer cancel2()
+
+	//nolint:errcheck
+	go dq.Decrypt(ctx2, []byte("12"), []byte("encrypted"))
+
+	waitForPendingRequestToBeAdded(t, dq, []byte("12"))
+
+	ctx3, cancel3 := context.WithTimeout(context.Background(), time.Duration(1000)*time.Millisecond)
+	defer cancel3()
+
+	//nolint:errcheck
+	go dq.Decrypt(ctx3, []byte("13"), []byte("encrypted"))
+
+	waitForPendingRequestToBeAdded(t, dq, []byte("13"))
+
+	requests := dq.GetRequests(4, 30)
+	expected := []DecryptionRequest{
+		{[]byte("11"), []byte("encrypted")},
+		{[]byte("12"), []byte("encrypted")},
+	}
+	if !reflect.DeepEqual(requests, expected) {
+		t.Error("GetRequests did not return the expected requests when totalBytesLimit is exceeded")
+	}
+}
+
+func TestGetRequestsWithShortPendingRequestQueue(t *testing.T) {
+	dq := NewThresholdDecryptor(4, 1000, 1000)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(1000)*time.Millisecond)
+	defer cancel()
+
+	//nolint:errcheck
+	go dq.Decrypt(ctx, []byte("11"), []byte("encrypted"))
+
+	waitForPendingRequestToBeAdded(t, dq, []byte("11"))
+
+	requests := dq.GetRequests(2, 1000)
+	expected := []DecryptionRequest{
+		{[]byte("11"), []byte("encrypted")},
+	}
+	if !reflect.DeepEqual(requests, expected) {
+		t.Error("GetRequests did not return the expected requests when limit is set")
+	}
+}
+
+func TestGetRequestsWithExpiredRequest(t *testing.T) {
+	dq := NewThresholdDecryptor(4, 1000, 1000)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(100)*time.Millisecond)
+	defer cancel()
+
+	//nolint:errcheck
+	dq.Decrypt(ctx, []byte("11"), []byte("encrypted"))
+
+	requests := dq.GetRequests(2, 1000)
+	expected := []DecryptionRequest{}
+	if !reflect.DeepEqual(requests, expected) {
+		t.Error("GetRequests did not return the expected requests when limit is set")
+	}
+}
+
+func TestStart(t *testing.T) {
+	dq := NewThresholdDecryptor(4, 1000, 1000)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := dq.Start(ctx)
+
+	if err != nil {
+		t.Error("Start returned unexpected error")
+	}
+}
+
+func TestClose(t *testing.T) {
+	dq := NewThresholdDecryptor(4, 1000, 1000)
+
+	dq.ResultReady([]byte("14"), []byte("decrypted"))
+
+	err := dq.Close()
+
+	if err != nil {
+		t.Error("Close returned unexpected error")
+	}
+}
+
+func waitForPendingRequestToBeAdded(t *testing.T, dq *decryptionQueue, ciphertextId CiphertextId) {
+	NewGomegaWithT(t).Eventually(func() bool {
+		dq.mu.Lock()
+		_, exists := dq.pendingRequests[string(ciphertextId)]
+		dq.mu.Unlock()
+		return exists
+	}, "15s", "10ms").Should(BeTrue(), "Pending request should be added")
```

Comment:
Do not use hard-coded timings like `15s` because this will soon become a flakey test.
Use `testutils.WaitTimeout` (search for usage in Core).